third-party/imgui/imgui_impl_glfw.cpp

```diff
@@ -238,7 +238,7 @@ void ImGui_ImplGlfw_Shutdown()
     ImGui::Shutdown();
 }
 
-void ImGui_ImplGlfw_NewFrame()
+void ImGui_ImplGlfw_NewFrame(float scale_factor)
```

Comment:
I'd also add boundaries check for the scale_factor to prevent improper API usage